'danger-credential-internals' without 'resident-key-support' results in compilation errors, is that intentional?

### I did this

Use the crate with feature 'danger-credential-internals'

### I expected the following

Standard compilation (with access to type conversions around Passkey/Credential)

### What actually happened

Compilation error:
Cannot find type `AttestedResidentKey` in this scope

### Version (and git commit)
master

### Operating System / Version
macos

### Any other comments

Specifically I am seeing this:
```
error[E0412]: cannot find type `AttestedResidentKey` in this scope
   --> /Users/ash/.cargo/git/checkouts/webauthn-rs-13492a96c2900e72/7696d8a/webauthn-rs/src/interface.rs:486:11
    |
486 | impl From<AttestedResidentKey> for Credential {
    |     -     ^^^^^^^^^^^^^^^^^^^ not found in this scope
    |     |
    |     help: you might be missing a type parameter: `<AttestedResidentKey>`
```

Replicated a handful of times. Seems the issue is that I have only the feature `danger-credential-internals` set, since I am trying to minimize dangerous features, but this is trying to define an impl definition on a type that doesn't exist unless `resident-key-support` is set. I am not familiar enough with rust to know if there is a way to have a code segment only exist if multiple feature flags are set.

I only want to use to use `danger-credential-internals`, but it doesn't currently seem feasible.

--- a/webauthn-rs/src/interface.rs
+++ b/webauthn-rs/src/interface.rs
@@ -480,14 +480,20 @@ impl PartialEq for AttestedResidentKey {
     }
 }
 
-#[cfg(all(feature = "danger-credential-internals", feature = "resident-key-support"))]
+#[cfg(all(
+    feature = "danger-credential-internals",
+    feature = "resident-key-support"
+))]
 impl From<AttestedResidentKey> for Credential {
     fn from(dk: AttestedResidentKey) -> Self {
         dk.cred
     }
 }
 
-#[cfg(all(feature = "danger-credential-internals", feature = "resident-key-support"))]
+#[cfg(all(
+    feature = "danger-credential-internals",
+    feature = "resident-key-support"
+))]
 impl From<Credential> for AttestedResidentKey {
     /// Convert a generic webauthn credential into a security key
     fn from(cred: Credential) -> Self {

--- a/webauthn-rs/src/interface.rs
+++ b/webauthn-rs/src/interface.rs
@@ -480,14 +480,14 @@ impl PartialEq for AttestedResidentKey {
     }
 }
 
-#[cfg(feature = "danger-credential-internals")]
+#[cfg(all(feature = "danger-credential-internals", feature = "resident-key-support"))]
 impl From<AttestedResidentKey> for Credential {
     fn from(dk: AttestedResidentKey) -> Self {
         dk.cred
     }
 }
 
-#[cfg(feature = "danger-credential-internals")]
+#[cfg(all(feature = "danger-credential-internals", feature = "resident-key-support"))]
 impl From<Credential> for AttestedResidentKey {
     /// Convert a generic webauthn credential into a security key
     fn from(cred: Credential) -> Self {

